compiler/rustc_codegen_ssa/src/mir/statement.rs

```diff
@@ -115,6 +115,21 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 self.codegen_coverage(&mut bx, coverage.clone());
                 bx
             }
+            mir::StatementKind::CopyNonOverlapping(box mir::CopyNonOverlapping {
+                ref src,
+                ref dst,
+                ref size,
+            }) => {
+              bx.memcpy(
+                dst,
+                todo!(),
+                src,
+                todo!(),
+                size,
+                todo!(),
+              );
```

Comment:
It's at this point where I'm not entirely sure what to do, since I can directly call this memcpy function or implement something somewhere else like the other branches.

Reply:
Either way is fine. Given that this arm is and is going to remain fairly short there is no good reason to extract it to somewhere else, but its not a huge deal if you end up doing it, either.